tests/integrations/subprocess/test_Subprocess__run.py

```diff
@@ -34,11 +36,30 @@ def test_simple_call_with_path_arg(mock_sub, capsys, tmp_path):
     assert capsys.readouterr().out == ""
 
 
-def test_simple_verbose_call(mock_sub, capsys):
+def test_simple_debug_call(mock_sub, capsys):
     "If verbosity is turned out, there is output"
-    mock_sub.command.verbosity = 2
+    mock_sub.command.logger = Log(verbosity=2)
 
     mock_sub.run(['hello', 'world'])
 
     mock_sub._subprocess.run.assert_called_with(['hello', 'world'])
-    assert capsys.readouterr().out == ">>> hello world\n"
+    assert capsys.readouterr().out == ">>> \n>>> Running Command:\n>>>     hello world\n"
+
+
+def test_simple_deep_debug_call(mock_sub, capsys):
+    "If verbosity is turned out, there is output"
+    mock_sub.command.logger = Log(verbosity=3)
+
+    mock_sub.run(["hello", "world"])
+
+    mock_sub._subprocess.run.assert_called_with(["hello", "world"])
+
+    expected_output = ">>> \n>>> Running Command:\n>>>     hello world\n>>> Environment:"
+    # some env vars (eg PS1) can contain line breaks...so this tries to replicate
+    # Log._log()'s functionality to print multi-line content with the appropriate preface.
+    for env_var, value in os.environ.items():
```

Comment:
As above, a mocked os.environ is needed here.

Reply:
Replaced `os.environ` with mock and a literal comparison.